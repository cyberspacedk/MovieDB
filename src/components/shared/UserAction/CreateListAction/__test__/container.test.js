import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import CreateListActionConnected from '../container';

describe('FavoritesContainer ', () => {
  const store = configureStore()({
    myLists: {
      ids: [1, 2, 3],
    },
    database: {
      lists: { 1: {} },
    },
  });

  store.dispatch = jest.fn();

  const wrapper = shallow(<CreateListActionConnected store={store} />);
  const container = wrapper.dive();
  const instance = container.instance();

  it('Should match its snapshot', () => {
    expect(shallowToJson(container)).toMatchSnapshot();
  });

  xit('should update the count by 1 when invoked by default', () => {
    expect(instance.state('visiblePop')).toBeFalsy();
    instance.showPopover();
    expect(container.state('visiblePop')).toBe('visible');
  });

  xit('Check is calling container method.', () => {
    const wrapper = mount(<CreateListActionConnected store={store} />);
    const container = wrapper.find('CreateListActionContainer').instance();
    jest.spyOn(container, 'showPopover');
    container.showPopover = jest.fn();
    container.showPopover();
    expect(container.showPopover).toHaveBeenCalled();
  });

  xit('Map state and dispatch to props', () => {
    expect(container.props()).toEqual(
      expect.objectContaining({
        favorites: expect.any(Array),
        getCreatedListRequest: expect.any(Function),
        addMovieToListRequest: expect.any(Function),
      }),
    );
  });
});
