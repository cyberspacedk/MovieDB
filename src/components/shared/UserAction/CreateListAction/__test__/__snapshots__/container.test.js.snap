// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesContainer  Should match its snapshot 1`] = `
<CreateListActionContainer
  addMovieToListRequest={[Function]}
  getCreatedListRequest={[Function]}
  myLists={
    Array [
      Object {},
      undefined,
      undefined,
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
